gulpfile.js

```diff
@@ -3,40 +3,75 @@ var deployDir = 'docs';
 
 var gulp = require('gulp');
 var clean = require('gulp-clean');
+var concat = require('gulp-concat');
 var jshint = require('gulp-jshint');
+var uglify = require('gulp-uglify');
+var babel = require('gulp-babel');
+var inject = require('gulp-inject');
+var minifyHtml = require('gulp-minify-html');
+var minifyCss = require('gulp-clean-css');
+var browserSync = require('browser-sync');
+var injectVersion = require('gulp-inject-version');
 
-gulp.task('default', ['copy']);
+gulp.task('default', ['lint', 'html']);
 
-gulp.task('copy', ['clean', 'lint'], function() {
-    return gulp.src(src)
-        .pipe(gulp.dest(deployDir));
+gulp.task('build', ['build-html']);
+
+gulp.task('clean', function () {
+  return gulp.src(deployDir)
+      .pipe(clean());
+});
+
+gulp.task('lint', function () {
+  var filesToLint = src.map(path => path + '/*.js');
+  filesToLint.push('gulpfile.js');
+
+  return gulp.src(filesToLint)
+      .pipe(jshint({
+        eqeqeq: true,
+        esversion: 6,
+        eqnull: true
+      }))
+      .pipe(jshint.reporter('jshint-stylish'))
+      .pipe(jshint.reporter('fail'));
 });
 
-gulp.task('clean', function() {
-    return gulp.src(deployDir)
-        .pipe(clean());
+gulp.task('build-js', ['clean'], function () {
+  return gulp.src(src.map(path => path + '/*.js'))
```

Comment:
Shouldn't the `sw/` directory be _inside_ `src/`? Also, I think `sw.js` is supposed injected into `index.html`. Only `sw/install.js` should be injected.